File: Geo-computing/python/lab0/lab0.py
#!/usr/bin/env python

import stopwatch  as sw
import dsp as dsp 
import numpy as np 

try: 
    import matplotlib.pyplot as plt
    plot=True
except:
    print "sorry, I won't be able to plot"
    plot=False



print "Python"

n = 100001
maxtime = 2.0 
a =0.9999 

x= np.zeros(n, dtype='Float32')

x[int(n/2)]= 1.0 
#x[0]=x[int(n/2)]= x[n-1]= 1.0 

nsmooth=0
y= np.zeros(n, dtype='Float32')

sw.start()
while sw.time() < maxtime :
    dsp.smooth(a,x,y)
    nsmooth+=1
sw.stop()

print "nsmooth = %d "%nsmooth
print "   mean = %9.7f "%dsp.mean(y)
print "   time = %9.7f "%sw.time()
print " mflops = %f "%((6.0e-6*n*nsmooth/sw.time()))



if plot:
    ll = plt.plot(range(n),y)
    plt.show()

File: Geo-computing/java/lab0/Lab0.java
package lab0;

/**
 * Benchmark recursive smoothing filter in Java.
 * @author Dave Hale, Colorado School of Mines
 * @version 2012.08.19
 */
public class Lab0 {
  public static void main(String[] args) {
    System.out.println("Java");
    int n = 1001;
    double maxtime = 2.0;
    float a = 0.99f;
    float[] x = new float[n];
    float[] y = new float[n];
    x[0] = x[n/2] = x[n-1] = 1.0f;
    int nsmooth;
    Stopwatch sw = new Stopwatch();
    sw.start();
    for (nsmooth=0; sw.time()<maxtime; ++nsmooth)
      Dsp.smooth(a,x,y);
    sw.stop();
    System.out.println("nsmooth = "+nsmooth);
    System.out.println("   mean = "+Dsp.mean(y));
    System.out.println("   time = "+sw.time());
    System.out.println(" mflops = "+(int)(6.0e-6*n*nsmooth/sw.time()));
  }
}
